pkg/proxy/handlers/spacelister.go

```diff
@@ -60,7 +62,9 @@ func (s *SpaceLister) ListUserWorkspaces(ctx echo.Context) ([]toolchainv1alpha1.
 	userID, _ := ctx.Get(context.SubKey).(string)
 	username, _ := ctx.Get(context.UsernameKey).(string)
 
-	signup, err := s.GetSignupFunc(userID, username)
+	c, _ := gin.CreateTestContext(httptest.NewRecorder())
```

Comment:
It doesn't look right to use a text context in runtime. Let's just make the context param optional.

Reply:
Done.  This does mean though that if the context is nil the updating feature won't work.